### PR TITLE
Don't show negative numbers of estimated samples

### PR DESCRIPTION
**Pull request by sduckett**
A floor of zero is practical when the question is "how many things more do I need to touch?". There is a zero that you'll see until we get a value calculated, but that seems more accurate than `Integer.MIN_VALUE` as a magic number.

Resolves [#162392809](https://www.pivotaltracker.com/story/show/162392809)

### CODE REVIEW

File: server/eclipse-project/src/main/java/us/freeandfair/corla/json/DoSDashboardRefreshResponse.java [hunk 1]
```diff
@@ -180,13 +180,11 @@ public static DoSDashboardRefreshResponse createResponse(final DoSDashboard dash
...
-          int estimated = Integer.MIN_VALUE;
+          int optimistic = 0;
+          int estimated = 0;
           audited_contests.put(cta.contest().id(), cta.reason());
 
-          // FIXME does looking up the ComparisonAudit by name make sense?
           for (final ComparisonAudit ca : ComparisonAuditQueries.matching(cta.contest().name())) {
-            // FIXME The dashboard looks funny here.
             optimistic = Math.max(optimistic, Math.max(0, ca.optimisticSamplesToAudit() - ca.getAuditedSampleCount()));
             estimated = Math.max(estimated, Math.max(0, ca.estimatedSamplesToAudit() - ca.getAuditedSampleCount()));
```
Comment (gws) [inline server/eclipse-project/src/main/java/us/freeandfair/corla/json/DoSDashboardRefreshResponse.java]: Can and should we reuse `ComparisonAudit.estimatedRemaining()` here?

File: server/eclipse-project/src/main/java/us/freeandfair/corla/json/DoSDashboardRefreshResponse.java [hunk 2]
```diff
@@ -180,13 +180,11 @@ public static DoSDashboardRefreshResponse createResponse(final DoSDashboard dash
...
-          int optimistic = Integer.MIN_VALUE;
-          int estimated = Integer.MIN_VALUE;
+          int optimistic = 0;
+          int estimated = 0;
           audited_contests.put(cta.contest().id(), cta.reason());
 
-          // FIXME does looking up the ComparisonAudit by name make sense?
           for (final ComparisonAudit ca : ComparisonAuditQueries.matching(cta.contest().name())) {
-            // FIXME The dashboard looks funny here.
             optimistic = Math.max(optimistic, Math.max(0, ca.optimisticSamplesToAudit() - ca.getAuditedSampleCount()));
```
Comment (gws) [inline server/eclipse-project/src/main/java/us/freeandfair/corla/json/DoSDashboardRefreshResponse.java]: If it was worth having a `ComparisonAudit.estimatedRemaining()` then would it be worth a similar `optimisticRemaining()` method which is where this logic moves?

File: server/eclipse-project/src/main/java/us/freeandfair/corla/json/DoSDashboardRefreshResponse.java (hunk 1 above)
Comment (sduckett) [reply server/eclipse-project/src/main/java/us/freeandfair/corla/json/DoSDashboardRefreshResponse.java]: Oh yes, we should.

File: server/eclipse-project/src/main/java/us/freeandfair/corla/json/DoSDashboardRefreshResponse.java (hunk 2 above)
Comment (sduckett) [reply server/eclipse-project/src/main/java/us/freeandfair/corla/json/DoSDashboardRefreshResponse.java]: Come to think of it, we don't show the optimistic number in this dashboard view.

**Comment by teaforthecat** [comment]
If this works great! I'm curious why the change to zero didn't work. I also see more that could be done,
like the for loop will always have only one ComparisonAudit, but it's not a problem.

**Pull request merged**